File: app/src/main/java/com/shuyu/apprecycler/chat/detail/ChatDetailContract.java
package com.shuyu.apprecycler.chat.detail;

import android.content.Context;

import java.util.List;

/**
 * 接口
 * Created by guoshuyu on 2017/9/4.
 */

public class ChatDetailContract {
    public interface IChatDetailView {
        void notifyView();

        void sendSuccess();

        Context getContext();
    }

    public interface IChatDetailPresenter<T> {

        void loadMoreData(int page);

        List<T> getDataList();

        List getMenuList();

        void sendMsg(String text);

        void sendMenuItem(int position);

    }
}

File: app-kotlin/src/main/java/example/shuyu/recycler/kotlin/chat/detail/ChatDetailContract.kt
package example.shuyu.recycler.kotlin.chat.detail

import android.content.Context

/**
 * 接口
 * Created by guoshuyu on 2017/9/4.
 */

class ChatDetailContract {
    interface IChatDetailView {

        val context: Context
        fun notifyView()

        fun sendSuccess()
    }

    interface IChatDetailPresenter<T> {

        val dataList: ArrayList<Any>

        val menuList: List<*>

        fun loadMoreData(page: Int)

        fun sendMsg(text: String)

        fun sendMenuItem(position: Int)

        fun release()

    }
}
